src/main/java/com/google/googleidentity/oauth2/token/UserClientTokenInfo.java

```diff
@@ -0,0 +1,92 @@
+/*
+    Copyright 2020 Google LLC
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+package com.google.googleidentity.oauth2.token;
+
+import com.google.common.io.BaseEncoding;
+
+import javax.crypto.Cipher;
+import java.nio.charset.StandardCharsets;
+import java.security.InvalidParameterException;
+import java.security.Key;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Used for encrypt username and clientID
+ */
+final class UserClientTokenInfo {
+    private final String username;
+    private final String clientID;
+    private final String tokenValue;
+
+    private static final Logger log = Logger.getLogger("UserClientTokenInfo");
+
+    private final static String delimiter = "\t";
+
+    UserClientTokenInfo(String username, String clientID, String tokenValue) {
+        this.username = username;
+        this.clientID = clientID;
+        this.tokenValue = tokenValue;
+    }
+
+    public String getUsername() {
+        return username;
+    }
+
+    public String getClientID() {
+        return clientID;
+    }
+
+    public String getTokenValue() {
```

Comment:
nit: add javadoc to explain that this string should not be directly returned to the client, and to see `getEncryptTokenString`